compiler/x/codegen/OpBinary.cpp

```diff
@@ -911,3 +911,904 @@ const TR_X86OpCode::TR_OpCodeBinaryEntry TR_X86OpCode::_binaryEncodings[IA32NumO
       {0x00, 0x00, 0x00, 0},    // AdjustFramePtr
       {0x00, 0x00, 0x00, 0}     // ReturnMarker
    };
+
+#include "codegen/FrontEnd.hpp"
+bool TR_X86OpCode::_UseDeprecatedBinaryEncoding = feGetEnv("TR_UseDeprecatedBinaryEncoding");
+// X87 is only available on X86-32
+#ifdef TR_TARGET_64BIT
+#define X87_________________(x) 0,0,0
+#else
+#define X87_________________(x) (uint8_t)((x & 0xE0) >> 5), (uint8_t)((x & 0x18) >> 3), (uint8_t)(x & 0x07)
+#endif
+
+const TR_X86OpCode::OpCode_t TR_X86OpCode::_binaries[] =
+{
+    { VEX_L___, VEX_vNONE, PREFIX___, REX__, ESCAPE_____, 0xcc, 0, ModRM_NONE, Immediate_0 },    // BADIA32Op/int 3
```

Comment:
There is a very important piece of implicit data for these instructions - namely the name of the instruction that each corresponds to... I'm wondering if we should try to make that more explicit some how - having information carried in comments is not a good way to make sure that information stays accurate and relevant.

One thought would be that since _binaries is an array it could be initialized as _binaries[numOpCodes] or something and each line would be something like:
_binaries[BADIAS32Op] = { .... } which would make the list not strictly dependent on order and it would enforce that the opcode was associated with the instance. I'm not sure if that incurs a penalty in initializing the array? I'm thinking not because the offsets are constant so it can all still be baked into the DLL.

Another option would be to add the op code to the encoding information, but that would increase footprint so I'm less of a fan of that.

@0dvictor would be interested to hear your thoughts. One additional benefit of the _binaries[<opcode>] format is that the instruction is on the left so you see it first before reading the encoding details which seems a more logical approach to me.